include/dqlite.h

```diff
@@ -218,6 +247,8 @@ int dqlite_node_enable_disk_mode(dqlite_node *n);
  */
 int dqlite_node_start(dqlite_node *n);
 
+int dqlite_node_handover(dqlite_node *n);
```

Comment:
Missing docstring.

The current implementation seems to not only handover the role, but also to stop the node, effectively makimg `dqlite_node_handover()` an alternative to `dqlite_node_stop()`. I feel this API should just handover the role, without stopping the node and leaving that responsibility to `dqlite_node_stop()`.

One reason is to give users a chance to first try to run `dqlite_node_handover()` and in case it returns non-zero possibly take appropriate measures **before** running `dqlite_node_stop()` and stop the node for good.

Another reason is that it somehow makes modular code harder: imagine that in the future we want to add some other optional behavior that happens before shutdown. We could add a `dqlite_node_some_new_pre_shutdown_behavior_that_also_stops_the_node()` function, which like `dqlite_node_handover` implements some specific pre-shutdown behavior and then stops the node. But what if we want to give users full control about choosing what pre-shutdown behavior to run? For instance they might want to run the new behavior **and** the handover behavior, or just one of them. The user might also have custom logic that they want to run between the handover and the actual stop.

If we separate concerns of optional pre-shutdown behavior and actual shutdown, we can have more control by mix and matching them, e.g.:

```C
int rv;

/* First try to handover our role */
rv = dqlite_node_handover(n);
if (rv != 0) {
    /* Run some custom logic that my application needs 
     * in case the handover fails. */
}

/* Also run the other new dqlite pre-shutdown behavior, in
 * this case we don't care about whether it succeeds or not */
dqlite_node_some_new_pre_shutdown_behavior(n);

/* Run app-specific logic that is required to run just before
* the final stop. */
my_app_pre_shutdown_behavior();

/* Finally stop the node for real */
dqlite_node_stop(n);
```

We should also make the timeout for `dqlite_node_handover()` configurable (it's kind if hard-coded right now): basically the function would return an error if it could not successfully handover the role within the specified timeout. We could have a reasonable default timeout, and let the users change it with something like `dqlite_node_set_handover_timeout(n, msecs)`. This could be done in a separate PR too.

Reply:
> Oh, I guess this isn't insurmountable, since the thread where dqlite_node_handover is called is not the same as the libuv main thread, so we can block in the "handover thread" on some signal from the libuv thread that the handover has wrapped up. Using pthread_condvar_t I suppose, or a semaphore (which is what dqlite_node_start does).

Exactly, the same approach implemented by `dqlite_node_start()` can be used here.

Reply:
Looks great, thanks!